File: BE/CleanArchitecture.Domain/Entities/Order.cs
using CleanArchitecture.Domain.Common;
using System.ComponentModel.DataAnnotations.Schema;

namespace CleanArchitecture.Domain.Entities;

public class Order : BaseEntity
{
    public int UserId { get; set; }
    public string OrderNumber { get; set; } = string.Empty;
    public string CustomerName { get; set; } = string.Empty;
    public string CustomerPhone { get; set; } = string.Empty;
    public string CustomerEmail { get; set; } = string.Empty;
    public string DeliveryAddress { get; set; } = string.Empty;
    public string Notes { get; set; } = string.Empty;
    public decimal TotalAmount { get; set; }

    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.PayOS;

    public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Pending;

    public string? PayOSOrderCode { get; set; }

    public DateTime? CompletedAt { get; set; }

    // Navigation properties
    public virtual User User { get; set; } = null!;
    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public virtual ICollection<WalletTransaction> WalletTransactions { get; set; } = new List<WalletTransaction>();
}

public enum OrderStatus
{
    Pending = 1,        // Chờ xử lý
    Confirmed = 2,      // Đã xác nhận
    Printing = 3,       // Đang in
    Shipping = 4,       // Đang giao
    Completed = 5,      // Hoàn thành
    Cancelled = 6       // Đã hủy
}

public enum PaymentMethod
{
    PayOS = 1,          // Thanh toán qua PayOS
    Wallet = 2          // Thanh toán từ ví
}

public enum PaymentStatus
{
    Pending = 1,        // Chờ thanh toán
    Paid = 2,           // Đã thanh toán
    Failed = 3,         // Thanh toán thất bại
    Refunded = 4        // Đã hoàn tiền
}

File: FE/src/types/order.ts
import { ProductType } from './product';

export enum OrderStatus {
  Pending = 1,
  Confirmed = 2,
  Printing = 3,
  Shipping = 4,
  Completed = 5,
  Cancelled = 6
}

export enum PaymentStatus {
  Pending = 1,
  Paid = 2,
  Failed = 3,
  Refunded = 4
}

export interface Order {
  id: number;
  userId: number;
  orderNumber: string;
  customerName: string;
  customerPhone: string;
  customerEmail: string;
  deliveryAddress: string;
  notes: string;
  totalAmount: number;
  status: OrderStatus;
  statusName: string;
  paymentStatus: PaymentStatus;
  createdAt: string;
  completedAt?: string;
  orderItems: OrderItem[];
}

export interface OrderItem {
  id: number;
  orderId: number;
  designId: number;
  productId: number;
  sizeWidth: number;
  sizeHeight: number;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  specialInstructions: string;
  designName: string;
  designPreviewUrl: string;
  designData: string; // JSON design session
  productName: string;
  productType: ProductType;
}

export interface CreateOrderDto {
  customerName: string;
  customerPhone: string;
  customerEmail: string;
  deliveryAddress: string;
  notes: string;
  orderItems: CreateOrderItemDto[];
}

export interface CreateOrderItemDto {
  designId: number;
  productId: number;
  sizeWidth: number;
  sizeHeight: number;
  quantity: number;
  specialInstructions: string;
}

export interface UpdateOrderStatusDto {
  status: OrderStatus;
}

export interface OrderStatusHistory {
  status: OrderStatus;
  statusName: string;
  timestamp: string;
  isCompleted: boolean;
  description: string;
}

export interface CartItem {
  id: number;
  designId: number;
  designName: string;
  designPreviewUrl: string;
  designData: string; // JSON design session
  productId: number;
  productName: string;
  productType: ProductType;
  sizeWidth: number;
  sizeHeight: number;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  specialInstructions: string;
  addedAt: string;
}

export interface AddToCartDto {
  designId: number;
  productId: number;
  sizeWidth: number;
  sizeHeight: number;
  quantity: number;
  specialInstructions: string;
}

export interface UpdateCartItemDto {
  quantity: number;
  specialInstructions: string;
}
